apps/emqx_durable_storage/src/emqx_ds_bitmask_keymapper.erl

```diff
@@ -201,29 +239,33 @@ make_keymapper(Bitsources) ->
         Bitsources
     ),
     {DimSizeof, Scanner} = lists:unzip(array:to_list(Arr)),
+    NDim = length(Scanner),
+    MaxOffsets = vec_max_offset(NDim, Bitsources),
     #keymapper{
         schema = Bitsources,
-        scanner = Scanner,
-        size = Size,
-        dim_sizeof = DimSizeof
+        vec_n_dim = length(Scanner),
+        vec_scanner = Scanner,
+        key_size = Size,
+        vec_coord_size = DimSizeof,
+        vec_max_offset = MaxOffsets
     }.
 
 -spec bitsize(keymapper()) -> pos_integer().
-bitsize(#keymapper{size = Size}) ->
+bitsize(#keymapper{key_size = Size}) ->
     Size.
 
 %% @doc Map N-dimensional vector to a scalar key.
 %%
 %% Note: this function is not injective.
--spec vector_to_key(keymapper(), vector()) -> key().
-vector_to_key(#keymapper{scanner = []}, []) ->
+-spec vector_to_key(keymapper(), vector()) -> scalar().
```

Comment:
Given the terminology note on l. 101, the name `vector_to_key` suggests it should return a `binary()`.  Perhaps this should be renamed to `vector_to_scalar`?

Reply:
True. But it may break _other branch_ once again, so I avoided it. I will add a comment for now.